src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -437,93 +433,62 @@ public class DosFileManager {
         return DosFileOperationResult.Error(ErrorCode.InvalidHandle);
     }
 
-    /// <summary>
-    /// Converts a dos filespec to a regex pattern
-    /// </summary>
-    /// <param name="fileSpec">The DOS filespec</param>
-    /// <returns>The regex pattern</returns>
-    private Regex FileSpecToRegex(string fileSpec) {
-        string regex = fileSpec.ToLowerInvariant();
-        regex = regex.Replace(".", "[.]");
-        regex = regex.Replace("?", ".");
-        regex = regex.Replace("*", ".*");
-        return new Regex(regex);
-    }
-
     private int? FindNextFreeFileIndex() {
-        for (int i = 0; i < _openFiles.Length; i++) {
-            if (_openFiles[i] == null) {
+        for (int i = 0; i < OpenFiles.Length; i++) {
+            if (OpenFiles[i] == null) {
                 return i;
             }
         }
 
         return null;
     }
 
-    private uint GetDiskTransferAreaAddressPhysical() {
-        return MemoryUtils.ToPhysicalAddress(_diskTransferAreaAddressSegment, _diskTransferAreaAddressOffset);
-    }
+    private uint GetDiskTransferAreaAddressPhysical() => MemoryUtils.ToPhysicalAddress(_diskTransferAreaAddressSegment, _diskTransferAreaAddressOffset);
 
     private OpenFile? GetOpenFile(ushort fileHandle) {
-        if (fileHandle >= _openFiles.Length) {
+        if (fileHandle >= OpenFiles.Length) {
             return null;
         }
-        return _openFiles[fileHandle];
+        return OpenFiles[fileHandle];
     }
 
-    private static bool IsValidFileHandle(ushort fileHandle) {
-        return fileHandle <= MaxOpenFiles;
-    }
+    private static bool IsValidFileHandle(ushort fileHandle) => fileHandle <= MaxOpenFiles;
 
     private DosFileOperationResult NoFreeHandleError() {
         if (_loggerService.IsEnabled(Serilog.Events.LogEventLevel.Warning)) {
             _loggerService.Warning("Could not find a free handle {MethodName}", nameof(NoFreeHandleError));
         }
         return DosFileOperationResult.Error(ErrorCode.TooManyOpenFiles);
     }
+    
+    /// <summary>
+    /// Converts dosFileName to a host file name.<br/>
+    /// For this, this needs to:
+    /// <ul>
+    /// <li>Prefix either the current folder or the drive folder.</li>
+    /// <li>Replace backslashes with slashes</li>
+    /// <li>Find case sensitive matches for every path item (since DOS is case insensitive but some OS are not)</li>
+    /// </ul>
+    /// </summary>
+    /// <param name="dosFileName">The file name to convert.</param>
+    /// <param name="forCreation">if true will try to find case sensitive match for only the parent of the file</param>
+    /// <returns>the file name in the host file system, or null if nothing was found.</returns>
+    internal string? ToHostCaseSensitiveFileName(string dosFileName, bool forCreation) => _dosPathResolver.ToHostCaseSensitiveFullName( dosFileName, forCreation);
```

Comment:
forCreation -> Although I think it was me who wrote that, I am not sure it is the best name, maybe convertParentOnly would be more representative?

Reply:
renamed to convertParentOnly.